### ISSUE TITLE
f# v5.0.0-preview.6 "The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment"

### ISSUE THREAD

**Issue by wallymathieu** [fsprojects/FSharpPlus#336]
### Description

Download .net [v5.0.0-preview.6](https://dotnet.microsoft.com/download/dotnet/5.0) and install

### Repro steps

1. `dotnet build -c Release FSharpPlus.sln`

### Expected behavior

Expected master branch to compile.

### Actual behaviour

```
Build FAILED.

/usr/local/share/dotnet/sdk/5.0.100-preview.6.20318.15/Microsoft.Common.CurrentVersion.targets(2084,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Runtime". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/TypeLevel/FSharpPlus.TypeLevel.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
    5 Warning(s)
    12 Error(s)

```
### Known workarounds

Using an earlier preview 5.0.100-preview.4.20258.7 or released versions of .net core to compile F#+.

### Related information 

* Operating system : Mac Os X
* Branch : Master
* .NET Runtime, CoreCLR or Mono Version : 5.0.100-preview.6.20318.15 
* Performance information, links to performance testing scripts

**Comment by wallymathieu** [comment]
Ping @dsyme @cartermp

**Comment by cartermp** [comment]
This only reproduces for me in this repo. For example, if I remove the `global.json` in XPlot, all the projects build with the latest SDK. 

I could reproduce the same issue and saw this at the top:

`/usr/local/share/dotnet/sdk/5.0.100-preview.6.20318.15/Microsoft.Common.CurrentVersion.targets(2084,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Runtime". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/Users/phillip/repos/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]`

that sounds to me like there's something in the build for this project causing it to go wrong. If all F# code on preview 6 couldn't find System.Runtime then no single F# project would build on that preview.

**Comment by wallymathieu** [comment]
If I remove the net45 from targetframewoks:
```xml
<TargetFrameworks>netstandard2.0</TargetFrameworks>
```
and comment out the system runtime reference 
```xml
 <Reference Condition=" '$(TargetFramework)' == 'net45'" Include="System.Runtime" />
```
you get the following:
```
mathieu@laszlo-lovasz FSharpPlus % dotnet build -c Release src/FSharpPlus
Microsoft (R) Build Engine version 16.7.0-preview-20310-07+ee1c9fd0c for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj (in 159 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]

Build FAILED.

/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): warning FS1116: A value marked as 'inline' has an unexpected value [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1114: The value 'FSharpPlus.Control.Sqrt.Sqrt' was marked inline but was not bound in the optimization environment [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(466,26): error FS1113: The value 'Sqrt' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/Control/Numeric.fs(463,9): error FS1118: Failed to inline the value 'Sqrt' marked 'inline', perhaps because a recursive value was marked 'inline' [/Users/mathieu/src/fs/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
    1 Warning(s)
    3 Error(s)
```

**Comment by wallymathieu** [comment]
I think this indicates that it's not related to `System.Runtime`.

**Comment by cartermp** [comment]
I just tried this on windows with preview 7 and the build was successful, let me try again on macOS since it's a new preview and thus a new F# compiler.

**Comment by cartermp** [comment]
Okay, confirmed this warning on my macOS machine in my fork here: https://github.com/cartermp/fsharpplus

However it did successfully compile:

```fsharp
λ ~/repos/FSharpPlus/ master dotnet build -c Release src/FSharpPlus
Microsoft (R) Build Engine version 16.7.0-preview-20360-03+188921e2f for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
/usr/local/share/dotnet/sdk/5.0.100-preview.7.20366.6/Microsoft.Common.CurrentVersion.targets(2084,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Runtime". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/Users/phillip/repos/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
  FSharpPlus -> /Users/phillip/repos/FSharpPlus/src/FSharpPlus/bin/Release/net45/FSharpPlus.dll
  FSharpPlus -> /Users/phillip/repos/FSharpPlus/src/FSharpPlus/bin/Release/netstandard2.0/FSharpPlus.dll

Build succeeded.

/usr/local/share/dotnet/sdk/5.0.100-preview.7.20366.6/Microsoft.Common.CurrentVersion.targets(2084,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Runtime". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [/Users/phillip/repos/FSharpPlus/src/FSharpPlus/FSharpPlus.fsproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.01
λ ~/repos/FSharpPlus/ master
```

**Comment by wallymathieu** [comment]
Sounds good, I'll try it out 😄

**Comment by wallymathieu** [comment]
It compiles on my machine as well 👍
